contents/ruby/Rails.md

```diff
@@ -0,0 +1,154 @@
+
+```c
+elements[1]
+```
+
+Rails developers can use ordinal representation to access the same element with the command
+
+```ruby
+elements.second
+```
+
+And while this may not reduce the amount one has to type, most Rails enthusiasts agree that it makes the code much more readable, intuitive, and inherently happiness-inducing.
+
+### Large and Active Community of Developers
+
+The Ruby language has a very passionate and extensive community of developers, and there is a vast array of libraries (called "Gems"), that can be used with your Rails projects. These libraries provide a lot of functionalities, and can be used to add flexibility to Rails projects when required. Furthermore, as a result of its popularity, it is very likely that any problems you encounter with Rails will have a solution that you can find on Google or StackOverflow. If not, the Ruby community is actually very famous for being 'nice', so you're likely to get a response to a new problem very quickly too. 
+
+However, Rails might not be best suited for all use cases, and there are certain areas where other frameworks might be more suitable. For instance:-
+
+### Drawbacks
+
+#### Lack of Flexibility
+
+As demonstrated above, Rails follows very strict conventions, that make life much easier for developers - or so Rails thinks. It might be true that there are some developers who find Rails' convention very stifling, and, as a result, prefer to use frameworks that allow them more flexibility. Rails will still allow you to do as you please, but since the effort of undo-ing the conventions makes the process longer and less enjoyable, it might be better to avoid Rails.
+
+#### Performance Time
+
+While newer releases have combatted this problem to a large degree, many of the old releases of Rails have had a reputation for being very bloated and slow to respond. Applications made using NodeJS, or purely client-side applications with a minimal backend, tend to have a much lower delay between an action and a response. While it should be noted that the vast majority of delays in applications occur due to poor optimization, and not the inherent speed of the framework, Rails still remains more bulky and slower than many of its competitors. 
+
+#### Learning Curve
+
+Rails' tendency to do things "automagically" can also be very confusing to new users, who might be more used to configuring routes and directory structures themselves. Furthermore, it becomes much harder to go from using Rails to a different framework, as the lack of conventions in other frameworks means you will have to learn some new topics to be proficient. You also have to learn the Ruby language, and it might be harder considering that most front-end applications are in JavaScript and its variants, and you will have to juggle multiple languages while you develop your application. 
```

Comment:
> Furthermore, it becomes much harder to go from using Rails to a different framework, as the lack of conventions in other frameworks means you will have to learn some new topics to be proficient.

I disagree - I actually think that having experience with Rails conventions makes you write better web applications, because they incorporate so many best practices. The learning curve here is about learning the conventions of Rails. As for other frameworks that require you to write things more explicitly, it's easier to reason about the application as a beginner / by reading the code. Whereas in a Rails application, you won't be able to find the glue code that links everything together.

Reply:
Thinking of replacing it with
` You may also find it challenging to switch to other frameworks, where you may be required to configure routes and write SQL queries yourself, instead of having the framework do it for you.` Thoughts?